pkg/vault/write.go

```diff
@@ -0,0 +1,29 @@
+package vault
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/api"
+	"go.opencensus.io/trace"
+)
+
+func (vc *vaultClient) Write(engine, key, value string) (*api.Secret, error) {
+	if vc.config.traceEnabled {
+		var span *trace.Span
+		vc.ctx, span = trace.StartSpan(vc.ctx, fmt.Sprintf("%s/Write", vc.config.tracePrefix))
+		defer span.End()
+	}
+
+	data := map[string]interface{}{
+		"data": map[string]interface{}{
+			key: value,
+		},
+	}
+
+	secret, err := vc.client.Logical().Write(engine, data)
+	if err != nil {
+		return secret, fmt.Errorf("failed to write data to %s: %w", engine, err)
+	}
+
+	return secret, nil
+}
```

Comment:
Do we have functionality in vault that checks for duplicate keys to ensure each one is unique? Just something that came to mind when thinking of writing new secrets.

eg. If someone tries to add a new secret with an existing key name, what would happen? It looks like it would overwrite an existing secret's value correct? Wondering if we should be sad-path testing to ensure people can't add duplicate keys (maybe this is something for osnp but figured I'd flag here).

Reply:
Great call out, check the test for writing to an existing secret.  From what I can tell, writing to a key that exists will create a new version and replace the existing key.  An alternative API could keep this write function private, and expose a create method that adds a check like the Update or Delete methods

Reply:
Updated the PR to make write a private function, only exposing the `Create, Update and Delete`